BaseInterfaces/test/runtests.jl

```diff
@@ -2,55 +2,16 @@ using BaseInterfaces
-    @test Interfaces.test(ArrayInterface, StepRange, [2:1:10])
-    @test Interfaces.test(ArrayInterface, Base.OneTo, [Base.OneTo(10)])
-    @test Interfaces.test(ArrayInterface, Base.Slice, [Base.Slice(100:150)])
-    @test Interfaces.test(ArrayInterface, Base.IdentityUnitRange, [Base.IdentityUnitRange(100:150)])
-    @test Interfaces.test(ArrayInterface, Base.CodeUnits, [codeunits("abcde")])
-    # No `getindex` defined for LogicalIndex
-    @test_broken Interfaces.test(ArrayInterface, Base.LogicalIndex, [to_indices([1, 2, 3], ([false, true, true],))[1]])
+# Test all interfaaces in BaseInterfaces
+@test Interfaces.test(BaseInterfaces)
 
-    # TODO test LinearAlgebra arrays and SparseArrays
-end
+# Or test each interface in the module individually
+@test Interfaces.test(ArrayInterface, BaseInterfaces)
+@test Interfaces.test(DictInterface, BaseInterfaces)
+@test Interfaces.test(IterationInterface, BaseInterfaces)
+@test Interfaces.test(SetInterface, BaseInterfaces)
 
-@testset "DictInterface" begin
-    @test Interfaces.test(DictInterface, Dict, [Arguments(d=Dict(:a => 1, :b => 2), k=:c, v=3)])
-    @test Interfaces.test(DictInterface, IdDict, [Arguments(d=IdDict(:a => 1, :b => 2), k=:c, v=3)])
-    @test Interfaces.test(DictInterface, Base.EnvDict, [Arguments(d=Base.EnvDict())])
-    @test Interfaces.test(DictInterface, Base.ImmutableDict, [Arguments(d=Base.ImmutableDict(:a => 1, :b => 2))])
-    @static if VERSION >= v"1.9.0"
-        @test Interfaces.test(DictInterface, Base.Pairs, [Arguments(d=Base.pairs((a = 1, b = 2)))])
-    end
-    @test Interfaces.test(DictInterface, Test.GenericDict, [Arguments(d=Test.GenericDict(Dict(:a => 1, :b => 2)), k=:c, v=3)])
-    GC.enable(false) # Avoid segfaults from garbage collection of WeakKeyDict keys
-    a = Ref(1); b = Ref(2); k=Ref(3)
-    @test Interfaces.test(DictInterface, WeakKeyDict, [Arguments(; d=WeakKeyDict(a => 1, b => 2), k, v=3)])
-    GC.enable(true)
-end
-
-@testset "IterationInterface" begin
-    @test Interfaces.test(IterationInterface, UnitRange, [1:5, -2:2])
-    @test Interfaces.test(IterationInterface, StepRange, [1:2:10, 20:-10:-20])
-    @test Interfaces.test(IterationInterface, Array, [[1, 2, 3, 4], [:a :b; :c :d]])
-    @test Interfaces.test(IterationInterface, Base.Generator, [(i for i in 1:5), (i for i in 1:5)])
-    @test Interfaces.test(IterationInterface, Tuple, [(1, 2, 3, 4)])
-end
-
-@testset "SetInterface" begin
-    @test Interfaces.test(SetInterface, Set, [Set((1, 2))])
-    @test Interfaces.test(SetInterface, BitSet, [BitSet((1, 2))])
-    @test Interfaces.test(SetInterface, Base.KeySet, [Base.KeySet(Dict(:a=>1, :b=>2))])
-    @test Interfaces.test(SetInterface, Test.GenericSet, [Test.GenericSet(Set((1, 2)))])
-    s = Base.IdSet(); push!(s, "a"); push!(s, "b")
-    @test Interfaces.test(SetInterface, Base.IdSet, [s]) 
-end
+@test_broken Interfaces.test(ArrayInterface, Base.LogicalIndex, [to_indices([1, 2, 3], ([false, true, true],))[1]])
```

Comment:
Is this fixable in Base Julia?

Reply:
Yes by putting `collect(LogicalIndex(I))` [here](https://github.com/JuliaLang/julia/blob/2a842146ac6ee0a7f505d56b4c1f77274d7f0628/base/indices.jl#L294C39-L294C39)

But of course that has a serious performance hit.

See: https://github.com/JuliaLang/julia/issues/51071